src/cpp/rtps/builtin/data/WriterProxyData.cpp

```diff
@@ -439,118 +443,122 @@ bool WriterProxyData::writeToCDRMessage(
     }
     if ( m_qos.m_durability.send_always() || m_qos.m_durability.hasChanged)
     {
-        if (!fastdds::dds::QosPoliciesSerializer<DurabilityQosPolicy>::add_to_cdr_message(m_qos.m_durability, msg))
+        if (!fastdds::dds::QosPoliciesSerializer<dds::DurabilityQosPolicy>::add_to_cdr_message(m_qos.m_durability, msg))
         {
             return false;
         }
     }
     if (m_qos.m_durabilityService.send_always() || m_qos.m_durabilityService.hasChanged)
     {
-        if (!fastdds::dds::QosPoliciesSerializer<DurabilityServiceQosPolicy>::add_to_cdr_message(m_qos.
+        if (!fastdds::dds::QosPoliciesSerializer<dds::DurabilityServiceQosPolicy>::add_to_cdr_message(m_qos.
```

Comment:
Can you please check also this function?
```suggestion
        if (!fastdds::dds::QosPoliciesSerializer<dds::DurabilityServiceQosPolicy>::add_to_cdr_message(
        m_qos.m_durabilityService, msg))
```